### ISSUE TITLE
PPG and PPI cannot be listened to for Verity Sense (iOS)

### ISSUE THREAD

**Issue by schmidtseb** [Rexios80/polar#5]
There is an issue with `startOhrStreaming` and `startOhrPPIStreaming` when trying to listen to their returned streams on iOS. They won't yield any events even though data is arriving from the sensor. I took a closer look into the issue regarding PPG and can now tell what is happening. Data from the sensor has a format according to `PolarOhrData`. Here, a field `type` of datatype `OhrDataType type` exists. The recorded data has a value of 4 in my case. Within `PolarOhrData.fromJson`, `OhrDataTypeExtension.fromJson` is called that should convert this integer to an `OhrDataType`. However, only 2 fields in this enum exist, so 4 is out of range. This is an error, but it is never thrown. So, from the outside it appears like no events are put on the stream, i.e. that the device is not recording data.
My quick and dirty fix is a change of `fromJson` in `ohr_data_type.dart`:
```
return OhrDataType.values[(json as int) - 4];
```

This does not make much sense, but it works for now. I would love to fix the issue by myself, but I am not completely familiar with the data. The 4 returned by the Verity Sense looks like the number of available channels. Maybe this is different for the OH1, though.

**Comment by Rexios80** [comment]
This should be fixed on develop. I don't have a device to test with, so try this in your pubspec and let me know if it works:
```yaml
polar:
  git:
    url: https://github.com/Rexios80/polar
    ref: 41fbf46d4efd8135873f880cada0274f20243d51
```

**Comment by schmidtseb** [comment]
This works great, thanks!

Since PPI is also not working, I also took a look into that. The Problem is within `polar_ohr_ppi_sample.dart`, where the format of the `json` data in the `fromJson` constructor isn't matching the structure of the class. This fixed the issue for me
```dart
  PolarOhrPpiSample.fromJson(Map<String, dynamic> json)
      : ppi = json['ppInMs'],
        errorEstimate = json['errorEstimate'],
        hr = json['hr'],
        blockerBit = json['blockerBit'] == 0 ? false : true,
        skinContactStatus = json['skinContactStatus'] == 0 ? false : true,
        skinContactSupported = json['skinContactSupported'] == 0 ? false : true;
```
ppi is named "ppInMs" and the flags are provided as integer and have to be converted to booleans!

**Comment by Rexios80** [comment]
Try this:
```yaml
polar:
  git:
    url: https://github.com/Rexios80/polar
    ref: 0b8023a6582638fe5cc3ce76cbf632efba7fd236
```

**Comment by Rexios80** [comment]
Try this:
```yaml
polar:
  git:
    url: https://github.com/Rexios80/polar
    ref: 8efa10067efe06f02a7e5e6787bfa61e69ffb177
```

**Comment by schmidtseb** [comment]
Now it works! Thanks a lot for your fast replies and changes!

**Comment by Rexios80** [comment]
Released in 2.1.0
